tactics/redexpr.ml

```diff
@@ -25,7 +25,8 @@ open Libobject
 (* call by value normalisation function using the virtual machine *)
 let cbv_vm env sigma c =
   if Coq_config.bytecode_compiler then
-    let ctyp = Retyping.get_type_of env sigma c in
+    (* Hack: drop the evarmap *)
+    let _sigma, ctyp = Typing.type_of env sigma c in
```

Comment:
This is insufficient, the named_context also needs checking.
also Typing doesn't check inside evar instances, but vm/native do reduce in the evar instances

Reply:
Honestly if we can't even trust the context we can just remove vm_compute.

Reply:
also don't forget to solve remaining constraints, it's probably possible to hide a type error in them

Reply:
Of course we can't trust the context, we can put stuff in it with `pose (x := ltac:(exact_no_check ...):...)`

Reply:
Yep, that's [wellknown: ltac variable bypasses typechecking](https://github.com/coq/coq/issues?q=label%3A%22wellknown%3A+ltac+variable+bypasses+typechecking%22+). I think the most blatant way to exploit it now might be `Ltac2.Constr.in_context`.

Reply:
>Yep, that's wellknown: ltac variable bypasses typechecking. 

No, that's yet another way to get ill typed terms.

pose is a regular variable not a ltac variable, typechecking is bypassed using no_check in my example.